src/extension.ts

```diff
@@ -69,7 +69,12 @@ export async function activate(context: vscode.ExtensionContext, perfStats: { lo
         activateContext.telemetry.properties.isActivationEvent = 'true';
         activateContext.telemetry.measurements.mainFileLoad = (perfStats.loadEndTime - perfStats.loadStartTime) / 1000;
 
-        ext.subscriptionProviderFactory = createVSCodeAzureSubscriptionProviderFactory();
+        // if this for a nightly test, we want to use the test subscription provider
+        const useAzureFederatedCredentials: boolean = !/^(false|0)?$/i.test(process.env['AzCode_UseAzureFederatedCredentials'] || '')
```

Comment:
very very very small nit, but this could just go into `getSubscriptionProviderFactory`.

Reply:
I only kept it here because it's also used to set `activateContext.errorHandling.rethrow `. Should this be set in the `getSubscriptionProviderFactory` instead? Should I create the `useAzureFederatedCredentials` boolean in both?

Reply:
Ah, in that case, I think you could just pass the activateContext into `getSubscriptionProviderFactory` instead and set it in there?

Reply:
Sure, done